target/linux/x86/base-files/lib/preinit/43_grow_rootfspart

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+get_squashfs_size() {
+	local low high
+	low=$(hexdump -s 40 -n 4 -e '1/4 "%08x" "\n"' "$1")
+	high=$(hexdump -s 44 -n 4 -e '1/4 "%08x" "\n"' "$1")
+	echo $((0x$high$low))
+}
+
+run_grow_root_part() {
+    [ "$FAILSAFE" != "true" ] || return
+
+    local diskdev squashfs_magic
+    local SQUASHFS_ALIGN=$((0x10000))
+
+    . /lib/upgrade/common.sh
+
+    if export_bootdevice && export_partdevice diskdev 0; then
+	diskdev=/dev/$diskdev
+        squashfs_magic=$(dd bs=4 count=1 if="${diskdev}2" 2>/dev/null)
+	[ "$squashfs_magic" = "hsqs" ] || return
+	squashfs_size=$(get_squashfs_size "${diskdev}2")
+	squashfs_size=$((squashfs_size + (SQUASHFS_ALIGN-1) & (~(SQUASHFS_ALIGN-1))))
+	[ "$squashfs_size" -gt 0 ] || return
+	rootfs_data_magic=$(hexdump -s $((squashfs_size+0x400)) -n 4 -e '1/4 "%08x" "\n"' "${diskdev}2")
+	[ "$rootfs_data_magic" != "f2f52010" ] || return # F2FS
```

Comment:
Again, [ = ] && return ?
Same two lines below.

Reply:
I normally program in a way that every command will not return false if it is not an error. That allows me to use "set -e" or 'trap "xxx" ERR'. But, if it is not clear, I can invert the logic to == && or simply use a verbose "if then". Which one would be better?

Reply:
Changed all [!=] || to [=] &&